src/test/kotlin/net/leanix/githubagent/services/WebhookEventServiceTest.kt

```diff
@@ -176,5 +143,100 @@ class WebhookEventServiceTest {
+
+        verify { webSocketService.sendMessage(any(), any<ManifestFileUpdateDto>()) }
+    }
+
+    @Test
+    fun `should handle manifest file in subdirectory`() {
+        val payload = """{
+            "repository": {
+                "name": "repo",
+                "full_name": "owner/repo",
+                "owner": {"name": "owner"},
+                "default_branch": "main"
+            },
+            "head_commit": {
+                "added": ["a/b/c/$MANIFEST_FILE_NAME"],
+                "modified": [],
+                "removed": []
+            },
+            "installation": {"id": 1},
+            "ref": "refs/heads/main"
+        }"""
+
+        webhookEventService.consumeWebhookEvent("PUSH", payload)
+
+        verify(exactly = 1) {
+            webSocketService.sendMessage(
+                "/events/manifestFile",
+                ManifestFileUpdateDto(
+                    "owner/repo",
+                    ManifestFileAction.ADDED,
+                    "content",
+                    "a/b/c/$MANIFEST_FILE_NAME"
+                )
+            )
+        }
+    }
+
+    @Test
+    fun `should handle push event with multiple added and modified files`() {
+        val payload = """{
+            "repository": {
+                "name": "repo",
+                "full_name": "owner/repo",
+                "owner": {"name": "owner"},
+                "default_branch": "main"
+            },
+            "head_commit": {
+                "added": ["custom/path/added1/$MANIFEST_FILE_NAME", "custom/path/added2/$MANIFEST_FILE_NAME"],
+                "modified": ["custom/path/modified/$MANIFEST_FILE_NAME"],
+                "removed": []
+            },
+            "installation": {"id": 1},
+            "ref": "refs/heads/main"
+        }"""
+
+        webhookEventService.consumeWebhookEvent("PUSH", payload)
+
+        verify(exactly = 1) {
+            webSocketService.sendMessage(
+                "/events/manifestFile",
+                ManifestFileUpdateDto(
+                    "owner/repo",
+                    ManifestFileAction.ADDED,
+                    "content",
+                    "custom/path/added1/$MANIFEST_FILE_NAME"
+                )
+            )
+        }
+
+        verify(exactly = 1) {
+            webSocketService.sendMessage(
+                "/events/manifestFile",
+                ManifestFileUpdateDto(
+                    "owner/repo",
+                    ManifestFileAction.ADDED,
+                    "content",
+                    "custom/path/added2/$MANIFEST_FILE_NAME"
+                )
+            )
+        }
+
+        verify(exactly = 1) {
+            webSocketService.sendMessage(
+                "/events/manifestFile",
+                ManifestFileUpdateDto(
+                    "owner/repo",
+                    ManifestFileAction.MODIFIED,
+                    "content",
+                    "custom/path/modified/$MANIFEST_FILE_NAME"
+                )
+            )
+        }
```

Comment:
I am not sure if the order is important here. but if it is, we should use `verifyOrder`

Reply:
You can group all like this also:
```
 verify(exactly = 1) {
  webSocketService.sendMessage(1) 
  webSocketService.sendMessage(2)
  webSocketService.sendMessage(3)
}
```